docs/source/performance.rst

```diff
@@ -14,272 +15,228 @@ Version |release| for version |version| of the CF conventions.
-             common destination grid defined by the field construct
-             'dst'.*
-		
-   >>> operator = fl[0].regrids(dst, method='conservative',
-   ...                          return_operator=True)
-   >>> fl_regridded = cf.FieldList(f.regrids(operator) for f in fl)
-   
+The performance of `cf` is largely a function of the performance of
+Dask. All of the techniques that Dask supports for `improving
+performance <https://docs.dask.org/en/stable/best-practices.html>`_
+apply, and performance parameters are generally be set via Dask's
+`configuration settings
+<https://docs.dask.org/en/stable/configuration.html>`_, but the value
+of the important :ref:`chunk size <Chunks>` is set through the `cf`
+API.
 
 ----
 
-.. _LAMA:
-
-**Large Amounts of Massive Arrays (LAMA)**
-------------------------------------------
-
-Data are stored and manipulated in a very memory efficient manner such
-that large numbers of constructs may co-exist and be manipulated
-regardless of the size of their data arrays. The limiting factor on
-array size is not the amount of available physical memory, but the
-amount of disk space available, which is generally far greater.
-
-The basic functionality is:
-
-* **Arrays larger than the available physical memory may be created.**
-
-  Arrays larger than a preset number of bytes are partitioned into
-  smaller sub-arrays which are not kept in memory but stored on disk,
-  either as part of the original file they were read in from or as
-  newly created temporary files, whichever is appropriate.  Therefore
-  data arrays larger than a preset size need never wholly exist in
-  memory.
+.. _Lazy-operations:
 
-* **Large numbers of arrays which are in total larger than the
-  available physical memory may co-exist.**
 
-  Large numbers of data arrays which are collectively larger than the
-  available physical memory may co-exist, regardless of whether any
-  individual array is in memory or stored on disk.
+**Lazy operations**
+-------------------
 
-* **Arrays larger than the available physical memory may be operated
-  on.**
+In general, all `cf` data operations (such as reading from disk,
+regridding, collapsing, subspacing, arithmetic, etc.) are managed by
+Dask and so are `evaluated lazily
+<https://en.wikipedia.org/wiki/Lazy_evaluation>`_, meaning that
+operations are not actually performed until the result is actually
+inspected by, for instance, creating a plot of the data, writing the
+data to disk, printing the array values, etc. When multiple operations
+are applied one after another, none of the operations are computed
+until the result of final one is requested.
+
+When the result of a stack of lazy operations is computed it is not
+cached, so if the result is requested again then the calculations are
+re-computed. However, a construct's `~cf.Field.persist` method can be
+used force the result to be retained in memory for fast future access.
```

Comment:
Should we mention caching of the first, second and last values towards providing the printable representations, or is that overkill? If you did think it a good idea, I think a note or similar here, or an extra sentence or two, would be a good place for it.

Reply:
I like the idea of mentioning this here, but I can't think of a way to succinctly describe the conditions under which the values cached "for free" from `cf.read` persist. We can dwell on it for later ...